### PR TITLE
Api to get process versions

### PR DESCRIPTION
**Pull request by abilpraju-aot**
# Issue Tracking

JIRA: 
Issue Type: BUG/ FEATURE
https://aottech.atlassian.net/browse/FWF-3580
image
[
    {
        "id": "3",
        "processName": "test1",
        "createdBy": "formsflow-designer",
        "created": "2024-09-11 04:24:37.210723",
        "majorVersion": 3,
        "minorVersion": 0
    },
    {
        "id": "2",
        "processName": "test1",
        "createdBy": "formsflow-designer",
        "created": "2024-09-11 04:24:28.593201",
        "majorVersion": 2,
        "minorVersion": 0
    },
    {
        "id": "1",
        "processName": "test1",
        "createdBy": "formsflow-designer",
        "created": "2024-09-11 04:24:05.186678",
        "majorVersion": 1,
        "minorVersion": 0
    }
]

### CODE REVIEW

File: forms-flow-api/src/formsflow_api/models/process.py [hunk 1]
```diff
@@ -136,3 +137,13 @@ def get_latest_version(cls, process_name):
         )
 
         return query
+
+    @classmethod
+    def fetch_histories_by_process_name(cls, process_name: str) -> List["Process"]:
```
Comment (sumesh-aot) [inline forms-flow-api/src/formsflow_api/models/process.py:142]: List["Process"]:
Having the Process class as type added is better than as that would make the typing, string value is for forward reference when you don't have access to the class yet

File: forms-flow-api/src/formsflow_api/models/process.py [hunk 2]
```diff
@@ -136,3 +137,13 @@ def get_latest_version(cls, process_name):
         )
 
         return query
+
+    @classmethod
+    def fetch_histories_by_process_name(cls, process_name: str) -> List["Process"]:
+        """Fetch all versions (histories) of a process by process_name."""
+        assert process_name is not None
+        query = (
```
Comment (sumesh-aot) [inline forms-flow-api/src/formsflow_api/models/process.py:145]: Shouldn't this add tenant authorization, or is it added somewere else ?

File: forms-flow-api/src/formsflow_api/resources/process.py [hunk 3]
```diff
@@ -232,3 +245,33 @@ def delete(process_id: int):
         """Delete process data by id."""
         response, status = ProcessService.delete_process(process_id), HTTPStatus.OK
         return response, status
+
+@cors_preflight("GET, OPTIONS")
+@API.route("/process-history/<string:process_name>", methods=["GET", "OPTIONS"])
```
Comment (sumesh-aot) [inline forms-flow-api/src/formsflow_api/resources/process.py:250]: For better resource reading, can we have the endpoint /process/<process_name>/versions ?

**Review by sumesh-aot — COMMENTED** [review]
Please add test cases

File: forms-flow-api/src/formsflow_api/models/process.py (hunk 1 above)
Comment (abilpraju-aot) [reply forms-flow-api/src/formsflow_api/models/process.py]: updated code

File: forms-flow-api/src/formsflow_api/models/process.py (hunk 2 above)
Comment (abilpraju-aot) [reply forms-flow-api/src/formsflow_api/models/process.py:145]: already included in auth_query

File: forms-flow-api/src/formsflow_api/resources/process.py (hunk 3 above)
Comment (abilpraju-aot) [reply forms-flow-api/src/formsflow_api/resources/process.py]: updated

**Review by arun-s-aot — APPROVED** [review]

**Review by sumesh-aot — APPROVED** [review]

**Pull request merged**